test/test_lyric.py

```diff
@@ -0,0 +1,205 @@
+            "sentence": "",
+            "start_tick": 300,
+            "end_tick": 780
+        }, new_line)
+        # Check if cloned and sorted properly
+        assert_lyric_lines_equal({
+            "sentence": "",
+            "start_tick": 300,
+            "end_tick": 780
+        }, lyrics.get_line_at_index(3))
+        assert_lyric_lines_equal({
+            "sentence": "",
+            "start_tick": 300,
+            "end_tick": 780
+        }, lyrics.get_line_at_index(4))
+
+    def test_clear(self):
+        lyrics = create_lyrics()
+        lyrics.create_line(10)
+        lyrics.clear()
+        self.assertEqual(len(lyrics._proto.lines), 0)
+
+
+class TestLyricLine(unittest.TestCase):
+    def test_create_word(self):
+        lyrics = create_lyrics()
+        line = lyrics.create_line(10)
+        line.create_word("first", 15, 20)
+        line.create_word("second", 30, 50)
+        self.assertEqual(len(line._proto.words), 2)
+        self.assertEqual(line._proto.words[0].word, "first")
+        self.assertEqual(line._proto.words[1].word, "second")
+
+    def test_clear(self):
+        lyrics = create_lyrics()
+        line = lyrics.create_line(10)
+        line.create_word("first", 15, 20)
+        line.create_word("second", 30, 50)
+        line.clear()
+        self.assertEqual(len(line._proto.words), 1)
+        self.assertEqual(line.is_empty(), True)
+
+    def test_get_index_of_line_at_tick(self):
+        lyrics = create_lyrics()
+        lyrics.create_line(10)
+        lyrics.create_line(20)
+        index_generator = lyrics.get_index_of_line_at_tick(10)
+        self.assertEqual(next(index_generator), 1)
+        self.assertEqual(next(index_generator), 0)
+
+    def test_remove_word_at_index(self):
+        lyrics = create_lyrics()
+        line = lyrics.create_line(9)
+        line.create_word("test", 15, 20)
+        # Replace the empty word
+        self.assertEqual(len(line._proto.words), 1)
+        line.remove_word_at_index(0)
+        # There will be 1 placeholder left
+        self.assertEqual(len(line._proto.words), 1)
+        assert_lyric_lines_equal({
+            "sentence": "",
+            "start_tick": 15,
+            "end_tick": 15 + DEFAULT_PPQ * 4 * 2
+        }, lyrics.get_line_at_index(1))
+
+    def test_remove_word(self):
+        lyrics = create_lyrics()
+        line = lyrics.create_line(10)
+        word = line.create_word("first", 15, 20)
+        line.remove_word(word)
+        self.assertEqual(len(line._proto.words), 1)
+        self.assertEqual(line.is_empty(), True)
+
+
+class TestLyricWord(unittest.TestCase):
+    def test_set_word(self):
+        lyrics = create_lyrics()
+        line = lyrics.create_line(10)
+        word = line._proto.words[0]
+        word = LyricWord(line, "test", 10, 20, word)
+        word.set_word("changed")
+        self.assertEqual(word.get_word(), "changed")
+
+    def test_set_start_tick(self):
+        lyrics = create_lyrics()
```

Comment:
Add another test for when we delete the word after changing start tick. Same for end tick.

Reply:
Please see the unresolved comment below.